### ISSUE TITLE
Problems in supervised MT

### ISSUE THREAD

**Issue by lihongzheng-nlp** [facebookresearch/XLM#279]
Hello @glample I'm running Chinese-Vietnamese(zh-vi) MT. I first pretrained the MLM model, then running the supervised model with following commands:
`python train.py --exp_name 'superMT_zh-vi' --dump_path 'dumped/' --reload_model 'dumped/zh-vi_tlm/rzpgproxlb/best-valid_mlm_ppl.pth,dumped/zh-vi_tlm/rzpgproxlb/best-valid_mlm_ppl.pth' --data_path 'data/zh-vi/' --lgs 'zh-vi' --mt_step 'zh-vi' --encoder_only false --emb_dim 512 --n_layers 6 --n_heads 8 --dropout '0.1' --attention_dropout '0.1' --gelu_activation true --tokens_per_batch 2000 --batch_size 32 --bptt 256 --optimizer 'adam_inverse_sqrt,beta1=0.9,beta2=0.98,lr=0.0001' --epoch_size 20000 --eval_bleu true --stopping_criterion 'valid_zh-vi_mt_bleu,10' --validation_metrics 'valid_zh-vi_mt_bleu'`

But it gave:

`data/zh-vi/train.vi-zh.vi.pth not found

data/zh-vi/train.vi-zh.zh.pth not found

data/zh-vi/valid.vi-zh.vi.pth not found

data/zh-vi/valid.vi-zh.zh.pth not found

data/zh-vi/test.vi-zh.vi.pth not found

data/zh-vi/test.vi-zh.zh.pth not found
p1:
 data/zh-vi/train.vi-zh.vi.pth
p2:
 data/zh-vi/train.vi-zh.zh.pth
p1:
 data/zh-vi/valid.vi-zh.vi.pth
p2:
 data/zh-vi/valid.vi-zh.zh.pth
p1:
 data/zh-vi/test.vi-zh.vi.pth
p2:
 data/zh-vi/test.vi-zh.zh.pth`

I was wondering that, **my translation direction is zh-vi, why the model needs train/valid/test.vi-zh.*** ?  Here're my data:

.
├── codes
├── test.vi.pth -> test.zh-vi.vi.pth
├── test.zh.pth -> test.zh-vi.zh.pth
├── test.zh-vi.vi
├── test.zh-vi.vi.pth
├── test.zh-vi.zh
├── test.zh-vi.zh.pth
├── train.vi
├── train.vi.pth
├── train.zh
├── train.zh.pth
├── valid.vi.pth -> valid.zh-vi.vi.pth
├── valid.zh.pth -> valid.zh-vi.zh.pth
├── valid.zh-vi.vi
├── valid.zh-vi.vi.pth
├── valid.zh-vi.zh
├── valid.zh-vi.zh.pth
├── vocab.vi
├── vocab.zh
└── vocab.zh-vi

And attached is the training log 
[superMT.train.log](https://github.com/facebookresearch/XLM/files/4332841/superMT.train.log)
. Are there anything wrong in my command line? Would you please help me solve the problems? Thank you very much!

**Comment by iConBarca** [comment]
I got this information from source code:
"Please ensure SRC < TGT.
This is just some convention to avoid the code to load both an English-French parallel dataset along with a French-English parallel dataset which would be exactly identical."
I got a same problem just like yours:
./data/processed/fr-en/train.en-fr.en.pth not found
./data/processed/fr-en/train.en-fr.fr.pth not found
./data/processed/fr-en/valid.en-fr.en.pth not found
./data/processed/fr-en/valid.en-fr.fr.pth not found
./data/processed/fr-en/test.en-fr.en.pth not found
./data/processed/fr-en/test.en-fr.fr.pth not found

And I don't know how to solve this problem, did you know how to do that? Ths.

**Comment by lihongzheng-nlp** [comment]
Hello @iConBarca as for my problem, I finally change the file names and solved, for example, I changed original name `train.zh-vi.vi.pth` to `train.vi-zh.vi.pth`. It seems that the code `lang2id` somewhere matters , but I didn't deep into it.

**Comment by renziver** [comment]
I'm having the same problem as well. I tried to rename from `train.<src>-<tgt>.en` into `train.<tgt>-<src>.en` but it's only working for some languages as I'm training a multilingual nmt. 

for some of the pairs, i've been requiring to have a format `train.<src>-<tgt>.en` while other pairs are `train.<tgt>-<src>.en`  which is odd.
